File: golang/qrcode.go
package novax

import (
	"net/http"
	"net/url"

	"github.com/shopspring/decimal"
)

// DynamicQrPayCreateRequest creates a dynamic QR-code pay order.
type DynamicQrPayCreateRequest struct {
	ReceiptOrderID string
	CompanyUserID  int64
	Currency       string
	Amount         decimal.Decimal
	Comment        string
}

func (r *DynamicQrPayCreateRequest) Method() string     { return http.MethodPost }
func (r *DynamicQrPayCreateRequest) Path() string       { return "/pay/v3/qr-code-pay/dynamic/create" }
func (r *DynamicQrPayCreateRequest) Extra() http.Header { return nil }
func (r *DynamicQrPayCreateRequest) Query() url.Values  { return nil }
func (r *DynamicQrPayCreateRequest) Body() any {
	return map[string]any{
		"receiptOrderId": r.ReceiptOrderID,
		"companyUserId":  r.CompanyUserID,
		"currency":       r.Currency,
		"amount":         r.Amount.String(),
		"comment":        r.Comment,
	}
}

File: rust/src/pay/qrcode.rs
use rust_decimal::Decimal;
use crate::request::{ApiRequest, HttpMethod};

pub struct DynamicQrPayCreateRequest {
    pub receipt_order_id: String,
    pub company_user_id:  i64,
    pub currency:         String,
    pub amount:           Decimal,
    pub comment:          Option<String>,
}

impl ApiRequest for DynamicQrPayCreateRequest {
    type Response = String;
    fn method(&self) -> HttpMethod { HttpMethod::Post }
    fn path(&self)   -> &str       { "/pay/v3/qr-code-pay/dynamic/create" }
    fn body(&self) -> Option<serde_json::Value> {
        Some(serde_json::json!({
            "receiptOrderId": self.receipt_order_id,
            "companyUserId":  self.company_user_id,
            "currency":       self.currency,
            "amount":         self.amount.to_string(),
            "comment":        self.comment,
        }))
    }
}
